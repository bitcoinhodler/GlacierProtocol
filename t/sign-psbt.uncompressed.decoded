{
  "txid": "4967238d112cbfa933c4f53f7a7993e907477410a90b02cd999787e2cd393e86",
  "hash": "4967238d112cbfa933c4f53f7a7993e907477410a90b02cd999787e2cd393e86",
  "version": 2,
  "size": 534,
  "vsize": 534,
  "weight": 2136,
  "locktime": 0,
  "vin": [
    {
      "txid": "66312654deda193b218dbe7fba08e1c4da444fc794d6b0f5fc66e306a92f61f2",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402207851298c1d40437ea81cf915bba05dd679736ac210992865392c388b5bcf600c02200f6c2bdca61d51b5977dba4ff414fa0cdcefc68072d47a9667b4a4a25f3e36b5[ALL] 3044022023fafbcc678bfa5815c046f5d7dda23793abc1d29524e45b496dd77bda40d0e902205f2242e5ec79dfc22430c6bb48e9a5f0a5711f36429849a56d97c899810f1ca4[ALL] 52410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354ae",
        "hex": "0047304402207851298c1d40437ea81cf915bba05dd679736ac210992865392c388b5bcf600c02200f6c2bdca61d51b5977dba4ff414fa0cdcefc68072d47a9667b4a4a25f3e36b501473044022023fafbcc678bfa5815c046f5d7dda23793abc1d29524e45b496dd77bda40d0e902205f2242e5ec79dfc22430c6bb48e9a5f0a5711f36429849a56d97c899810f1ca4014d0b0152410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00040660,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 762fea3246dc4ce19cfd3f1aa9685e8838605057 OP_EQUAL",
        "hex": "a914762fea3246dc4ce19cfd3f1aa9685e883860505787",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N42986nuqGfmbRUuojYCboE7EE9VLmYai2"
        ]
      }
    },
    {
      "value": 0.00004000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 fb5b6788a336c0cf49b90e51ad426f2e89bf43fb OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a914fb5b6788a336c0cf49b90e51ad426f2e89bf43fb88ac",
        "reqSigs": 1,
        "type": "pubkeyhash",
        "addresses": [
          "n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR"
        ]
      }
    }
  ]
}
