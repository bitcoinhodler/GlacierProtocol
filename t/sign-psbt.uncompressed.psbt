cHNidP8BAHUCAAAAAbO5l9wW38mEuJYivkSVExvrpLdZGh03PmRd486lvjFoAQAAAAD/////AtSeAAAAAAAAF6kUdi/qMkbcTOGc/T8aqWheiDhgUFeHoA8AAAAAAAAZdqkU+1tniKM2wM9JuQ5RrUJvLom/Q/uIrAAAAAAAAQDfAgAAAAG/P8jHZkaYw5NuSWryOwJyrlb6YRdwOovgzzUP3WeYPwEAAABqRzBEAiA8+TTsFM5aJsFU3dEzFFjIuLwtDtj/95fSjanNWulnBAIgRmYlU6FuEQm8jCh9AcS27K6HYbnHossIrdXHFipCUooBIQJzms+rOzwBjsnwkSWpOMg5lqyzXhbw/zHvuzZ3esv5fv////8CMAz1BQAAAAAZdqkUUp6ESGGs1sXpE9uBQhQqXkWAbBWIrFDDAAAAAAAAF6kUdi/qMkbcTOGc/T8aqWheiDhgUFeHAAAAAAEE/QsBUkEEd/FfIq7/rz87xIooCnZ/fGryEnZ4PAn/O8qrvs4XgRNc8PKPZD08YKdcKdsIGdVEK66D2PdOfETO0yVWZZmOUkEEA/ZCIjvHUeElq2BO0gxUTpM+uCoZIB+QvaYjap3b7ayTHBCMfDGJt/x+pwl7AiF9gZ6GaphBKRxSL2kXOhJSSkEEaJXz7QHAkCW6nPq3tn1pnj66zq2gvv9wvHLH8Q3y0cZ033Q7u1jOohFwgxM05NbZZe00ziMklVcJH0QDBaucAEEEnzyxUhMMJB4BuSIKFH9p9xEbgw3vNbUQlzq0e7mvf/byH8talYg5/uACLo6l3cbXH8YkqEvcQoNRk6v+4SWxw1SuAAEA/QsBUkEEd/FfIq7/rz87xIooCnZ/fGryEnZ4PAn/O8qrvs4XgRNc8PKPZD08YKdcKdsIGdVEK66D2PdOfETO0yVWZZmOUkEEA/ZCIjvHUeElq2BO0gxUTpM+uCoZIB+QvaYjap3b7ayTHBCMfDGJt/x+pwl7AiF9gZ6GaphBKRxSL2kXOhJSSkEEaJXz7QHAkCW6nPq3tn1pnj66zq2gvv9wvHLH8Q3y0cZ033Q7u1jOohFwgxM05NbZZe00ziMklVcJH0QDBaucAEEEnzyxUhMMJB4BuSIKFH9p9xEbgw3vNbUQlzq0e7mvf/byH8talYg5/uACLo6l3cbXH8YkqEvcQoNRk6v+4SWxw1SuAAA=
