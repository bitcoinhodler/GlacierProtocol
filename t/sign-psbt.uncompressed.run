#!/bin/bash
set -e

# Also tests excessive fee correction

$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.uncompressed.psbt
y
2
92jodMpEwGtS11sMjxfq6acnc1Yi6VX2QFZWr4YQVjXnQGdKH1g
922F7weRdtBgCnqcnuMo8hba8wj75fZc2nx4oTnLHevh9GaCVXA
y
INPUT
