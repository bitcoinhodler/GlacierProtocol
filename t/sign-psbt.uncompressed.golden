Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.00050000 BTC in unspent inputs from cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00040660 BTC going back to cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00004000 BTC going to destination address n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR
Fee amount: 0.00005340
Confirm? (y/n): 
How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00050000 BTC in unspent inputs from cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00040660 BTC going back to cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00004000 BTC going to destination address n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR
Fee amount: 0.00005340

Signing with private keys: 
92jodMpEwGtS11sMjxfq6acnc1Yi6VX2QFZWr4YQVjXnQGdKH1g
922F7weRdtBgCnqcnuMo8hba8wj75fZc2nx4oTnLHevh9GaCVXA


Confirm? (y/n): 
Calculating transaction...


Raw signed transaction (hex):
0200000001f2612fa906e366fcf5b0d694c74f44dac4e108ba7fbe8d213b19dade5426316601000000fd9f010047304402207851298c1d40437ea81cf915bba05dd679736ac210992865392c388b5bcf600c02200f6c2bdca61d51b5977dba4ff414fa0cdcefc68072d47a9667b4a4a25f3e36b501473044022023fafbcc678bfa5815c046f5d7dda23793abc1d29524e45b496dd77bda40d0e902205f2242e5ec79dfc22430c6bb48e9a5f0a5711f36429849a56d97c899810f1ca4014d0b0152410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354aeffffffff02d49e00000000000017a914762fea3246dc4ce19cfd3f1aa9685e883860505787a00f0000000000001976a914fb5b6788a336c0cf49b90e51ad426f2e89bf43fb88ac00000000

Transaction fingerprint (md5):
2a0ef20eb144bde2a03d5f50ae19ae64
QR code for transaction written to transaction.png
